spec/services/reports/bank_transactions/bank_transaction_report_creator_spec.rb

```diff
@@ -0,0 +1,78 @@
+require 'rails_helper'
+
+module Reports
+  module BankTransactions
+    RSpec.describe BankTransactionReportCreator do
+      let(:legal_aid_application) do
+        create :legal_aid_application,
+               :with_proceeding_types,
+               :with_everything,
+               :with_benefits_transactions,
+               :with_uncategorised_credit_transactions,
+               :with_cfe_v2_result,
+               :generating_reports
+      end
+
+      let(:remarks) { CFE::Remarks.new(populated_remarks_hash) }
+      let(:remarked_transaction) { double CFE::RemarkedTransaction, reasons: %i[amount_variation unknown_frequency] }
+      before do
+        allow(legal_aid_application.cfe_result).to receive(:remarks).and_return(remarks)
+        allow(remarks.review_transactions.transactions).to receive(:[]).and_call_original
+        allow(remarks.review_transactions.transactions).to receive(:[]).with(legal_aid_application.bank_transactions.first.id).and_return(remarked_transaction)
+      end
+
+      describe '.call' do
+        context 'with no local_csv param' do
+          subject { described_class.call(legal_aid_application) }
+
+          it 'attaches bank_transaction_report.csv to the application' do
+            expect_any_instance_of(CCMS::Requestors::ReferenceDataRequestor).to receive(:call)
+            subject
+            legal_aid_application.reload
+            expect(legal_aid_application.bank_transaction_report.document.content_type).to eq('text/csv')
+            expect(legal_aid_application.bank_transaction_report.document.filename).to eq('bank_transaction_report.csv')
+          end
+
+          it 'does not attach a report if one already exists' do
+            create :attachment, :bank_transaction_report, legal_aid_application: legal_aid_application
+            expect { subject }.not_to change { Attachment.count }
+          end
+        end
+
+        context 'with local_csv param' do
+          subject { creator.call(local_csv: true) }
+
+          let(:creator) { BankTransactionReportCreator.new(legal_aid_application) }
+          let(:tempfile) { Rails.root.join('tmp', 'bank_transactions.csv') }
+
+          it 'creates a local file' do
+            File.unlink(tempfile) if File.exist?(tempfile)
```

Comment:
Cool :+1: 
Not seen `unlink` before, I've always used a block and hoped it deleted it properly! 😄